packages/x-date-pickers/src/DateCalendar/tests/DateCalendar.test.tsx

```diff
@@ -637,8 +637,11 @@ describe('<DateCalendar />', () => {
       );
 
       const renderCountBeforeChange = RenderCount.callCount;
+      // TODO: Use userEvent.click instead.
+      fireEvent.focus(screen.getByRole('gridcell', { name: '2' }));
       fireEvent.click(screen.getByRole('gridcell', { name: '2' }));
-      expect(RenderCount.callCount - renderCountBeforeChange).to.equal(4); // 2 render * 2 days
+      // 2 render (one to update tabIndex + autoFocus, one to update selection) * 2 days * 2 (because dev mode)
+      expect(RenderCount.callCount - renderCountBeforeChange).to.equal(8);
```

Comment:
There is no additional render caused by this PR.
On master, we don't focus in the test and we only have 4 renders, but it's not a realistic scenario, clicking always focused and with focus with have 8 render before and after the changes.

Reply:
Makes sense. 👍